### PR TITLE
feat: add BackendConfig configuration for GKE

### PR DESCRIPTION
**Pull request by nakatamixi**
Hi!

This PR wants to add configuration about [CRD BackendConfig in Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine/docs/concepts/backendconfig) to argocd-server.

Checklist:

* [x] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed the CLA and the build is green.
* [x] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by nakatamixi** [comment]
I tested with additional values and It seems work well.

```
server:
  service:
    annotations:
      beta.cloud.google.com/backend-config: '{"default": "argocd-server"}'
  backendConfig:
    enabled: true
    spec:
      iap:
        enabled: true
        oauthclientCredentials:
          secretName: argocd-secret
configs:
  secret:
    extra:
      client_id: CLIENT_ID
      client_secret: CLIENT_SECRET
```

File: charts/argo-cd/README.md [hunk 1]
```diff
@@ -183,6 +183,8 @@ Helm v3 has removed the `install-crds` hook so CRDs are now populated by files i
 | server.autoscaling.maxReplicas | Maximum number of replicas for the server [HPA](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) | `5` |
 | server.autoscaling.targetCPUUtilizationPercentage | Average CPU utilization percentage for the server [HPA](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) | `50` |
 | server.autoscaling.targetMemoryUtilizationPercentage | Average memory utilization percentage for the server [HPA](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) | `50` |
+| server.backendConfig.enabled | Enable BackendConfig custom resource for Google Kubernetes Engine. | `false` |
+| server.backendConfig.spec | [BackendConfigSpec](https://cloud.google.com/kubernetes-engine/docs/concepts/backendconfig#backendconfigspec_v1beta1_cloudgooglecom) | `{}` |
```
Comment (seanson) [inline charts/argo-cd/README.md:187]: Could we prepend this variable name with `GKE`? I'd like to make it explicit in any kind of config that we're setting a Google specific variable like `server.GKEbackendConfig`

Comment (nakatamixi) [reply charts/argo-cd/README.md]: Thanks for kind review.
I fixed. 7a9c0e4

**Comment by nakatamixi** [comment]
Thanks a lot!

**Comment by redbaron** [comment]
@nakatamixi ,

what do you think of enabling it automatically ? I prepared draft PR https://github.com/argoproj/argo-helm/pull/396 , let me know what you think pls.

**Pull request merged**